fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/v1/ServerChannelEvents.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.networking.v1;
+
+import java.util.List;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Offers access to events related to the registration of network channels for a server-side network handler.
+ */
+public final class ServerChannelEvents {
+	/**
+	 * An event for the server play network handler receiving an update indicating the connected client's ability to receive packets in certain channels.
+	 *
+	 * @see PlayPacketSender#hasChannel(Identifier)
+	 */
+	public static final Event<Registered> REGISTERED = EventFactory.createArrayBacked(Registered.class, callbacks -> (handler, server, sender, channels) -> {
+		for (Registered callback : callbacks) {
+			callback.onChannelRegistered(handler, server, sender, channels);
+		}
+	});
+
+	/**
+	 * An event for the server play network handler receiving an update indicating the connected client's lack of ability to receive packets in certain channels.
+	 *
+	 * @see PlayPacketSender#hasChannel(Identifier)
+	 */
```

Comment:
Similar to the comment I made on the client side, more specifics about when this occurs would be good in the javadoc